### PR TITLE
Add withUnretained for Element == Void

### PR DESCRIPTION
**Pull request by MortyMerr**
https://github.com/RxSwiftCommunity/RxSwiftExt/issues/178

### CODE REVIEW

**Comment by freak4pc** [comment]
@MortyMerr please resolve conflicts.

Also, can you explain what's wrong with the existing impl.? It doesn't work for Void because of the resultSelector ?

**Comment by MortyMerr** [comment]
Hey @freak4pc sure. The purpose of the initial issue was to ignore Void element, only passing `self` parameter to result selector. I guess just for better syntax

File: Source/RxSwift/withUnretained.swift [hunk 1]
```diff
@@ -46,6 +46,32 @@ extension ObservableType {
...
+     Provides an unretained, safe to use (i.e. not implicitly unwrapped), reference to an object replacing the element emitted by the sequence in case its `Void`.
+     In the case the provided object cannot be retained successfully, the seqeunce will complete.
+
+     - parameter obj: The object to provide an unretained reference on.
+     - parameter resultSelector: A function to transform the unretained referenced on `obj`.
+     - returns: An observable sequence that contains the result of `resultSelector` being called with an unretained reference on `obj`.
+     */
+    public func withUnretained<T: AnyObject, Out>(_ obj: T, resultSelector: @escaping (T) -> Out)
+        -> Observable<Out> {
+            return withUnretained(obj) { object, _ in resultSelector(object) }
```
Comment (freak4pc) [inline Source/RxSwift/withUnretained.swift]:
```suggestion
        return withUnretained(obj) { object, _ in resultSelector(object) }
```

**Comment by MortyMerr** [comment]
We can change name to something like `unretained`
This repo has few operators which save just few symbols. But they are used so often, that it makes sense to use them. For instance, `mapTo(1)` vs `map { _ in 1 }`. This is just sugar which improves code readability.

`withUnretained(self).subscribe(onNext: { $0.doSomething })` looks better for me than `withUnretained(self).subscribe(onNext: { (sself, _) in sself.doSomething })`
But of course, it's really subjective.

> there's no specific subscribe / drive / emit for Void-emitting objects
Yes, but this repo extends the main repo functionality, so I don't think it's a valid argument to not add this sugar.

Shortly, I think it might be useful. However, I have no personal interest in this operator, just working on closing issues and fulfilling user requests. If it's not useful, feel free to close this PR & initial issue

**Comment by freak4pc** [comment]
> Yes, but this repo extends the main repo functionality, so I don't think it's a valid argument to not add this sugar.

My point was that even Ext doesn't aim to add these things. I find it a bit strange and adds relatively little value. Yes, this repo extends things, but every piece of code means more maintenance and a larger code base, so it needs to be things that have a solid use case and are highly useful and intuitive. 

I think this isn't the case here. I'll be closing for now but let me know if you have any other thoughts / ideas around this.

**Pull request closed without merging**